optaplanner-core/src/main/java/org/optaplanner/core/impl/heuristic/move/AbstractMove.java

```diff
@@ -33,9 +33,14 @@
     @Override
     public final AbstractMove<Solution_> doMove(ScoreDirector<Solution_> scoreDirector) {
         AbstractMove<Solution_> undoMove = createUndoMove(scoreDirector);
+        doMoveOnly(scoreDirector);
+        return undoMove;
+    }
+
+    @Override
+    public void doMoveOnly(ScoreDirector<Solution_> scoreDirector) {
```

Comment:
This method should be final here, like doMove is:
```suggestion
    public final void doMoveOnly(ScoreDirector<Solution_> scoreDirector) {
```

We don't want to allow users to overwrite this or think they need to implement this.